web/angular-wallet/src/app/network/network.service.ts

```diff
@@ -58,4 +59,13 @@ export class NetworkService {
   public addBlock(block: Block) {
     this.setBlocks([block].concat(this.blocks.value));
   }
+
+  public async isMainNet(): Promise<boolean> {
+    try {
+      const {previousBlockHash} = await this.api.block.getBlockByHeight(constants.mainnetIndicator.block, false);
```

Comment:
too simple...

Reply:
Might also be a mainnet out of sync. So perhaps combine it with another check if network is in sync. Like if the latest block is more than 8 minutes late, then you are probably out of sync.